sec_certs_page/user/views.py

```diff
@@ -0,0 +1,464 @@
+            new_user = User.create(
+                username=form.username.data,
+                email=form.email.data,
+                password=form.password.data,
+                roles=[]
+            )
+            # Send confirmation email
+            token = User.generate_confirmation_token(new_user.username)
+            confirmation_url = url_for('user.confirm_email', token=token, _external=True)
+            
+            msg = Message(
+                subject="Confirm your email - sec-certs",
+                recipients=[new_user.email],
+                html=render_template("user/emails/confirm_email.html.jinja2", 
+                                   user=new_user, confirmation_url=confirmation_url),
+                sender=current_app.config.get('MAIL_DEFAULT_SENDER')
+            )
+            
+            try:
+                mail.send(msg)
+                flash("Registration successful! Please check your email to confirm your account.", "success")
+                return redirect(url_for("user.login"))
+            except Exception as e:
+                flash("Registration successful but email could not be sent. Please contact support.", "warning")
+                return redirect(url_for("user.login"))
+        except ValueError as e:
+            flash(str(e), "error")
+            return render_template("user/register.html.jinja2", form=form)
+    
+    return render_template("user/register.html.jinja2", form=form)
+
+
+@user.route("/confirm-email/<token>")
+def confirm_email(token):
+    user_id = User.verify_token(token, 'email_confirmation')
+    if user_id:
+        user_obj = User.get(user_id)
+        if user_obj:
+            user_obj.confirm_email()
+            flash("Email confirmed successfully! You can now log in.", "success")
+        else:
+            flash("Invalid confirmation link.", "error")
+    else:
+        flash("Invalid or expired confirmation link.", "error")
+    
+    return redirect(url_for("user.login"))
+
+
+@user.route("/forgot-password", methods=["GET", "POST"])
+@register_breadcrumb(user, ".forgot_password", "Forgot Password")
+def forgot_password():
+    if current_user.is_authenticated:
+        return redirect(url_for("index"))
+    
+    form = PasswordResetRequestForm()
+    if form.validate_on_submit():
+        user_obj = User.get_by_email(form.email.data)
+        if user_obj:
+            token = User.generate_password_reset_token(user_obj.username)
+            reset_url = url_for('user.reset_password', token=token, _external=True)
+            
+            msg = Message(
+                subject="Password Reset - sec-certs",
+                recipients=[user_obj.email],
+                html=render_template("user/emails/reset_password.html.jinja2", 
+                                   user=user_obj, reset_url=reset_url),
+                sender=current_app.config.get('MAIL_DEFAULT_SENDER')
+            )
+            
+            try:
+                mail.send(msg)
```

Comment:
This and all of the other email sends should be handled asynchronously via dramatiq tasks. Just like how the notification emails are handled in the notifications blueprint.